### PR TITLE
Add hotspot rule S4825(Http Requests)

### LINKED ISSUE #1138: Rule S4825: Sending HTTP requests is security-sensitive
Implements [RSPEC-4825](https://jira.sonarsource.com/browse/RSPEC-5089)

### PR DESCRIPTION
**Pull request by andrea-guarino-sonarsource**
Fixes #1138

### CODE REVIEW

File: eslint-bridge/src/rules/http-requests.ts [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+let openCalls: Callee[] = [];
+let isHttpRequestConstructorCalled = false;
+
+export const rule: Rule.RuleModule = {
+  create(context: Rule.RuleContext) {
+    return {
+      Program() {
+        // init flag for each file
+        openCalls = [];
+        isHttpRequestConstructorCalled = false;
```
Comment (vilchik-elena) [inline eslint-bridge/src/rules/http-requests.ts]: `isHttpRequestObjectCreated`

File: eslint-bridge/src/rules/http-requests.ts [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+          openCalls.forEach(callee => context.report({ message: MESSAGE, node: callee }));
+        }
+      },
+    };
+  },
+};
+
+function checkCallExpression({ callee }: estree.CallExpression, context: Rule.RuleContext) {
+  // check for fetch API and jQuery API
+  if (isGlobalIdentifier(callee, "fetch") || isMemberExpression(callee, "$", ...jQueryFunctions)) {
```
Comment (vilchik-elena) [inline eslint-bridge/src/rules/http-requests.ts]: I think `jQuery` should be considered along with `$`

File: eslint-bridge/src/rules/http-requests.ts [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+// https://jira.sonarsource.com/browse/RSPEC-2077
```
Comment (andrea-guarino-sonarsource) [inline eslint-bridge/src/rules/http-requests.ts]: Update link

File: eslint-bridge/tests/rules/http-requests.test.ts [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+      code: `
+        var xmlhttp = new XMLHttpRequest();
+        xmlhttp.open("GET", url, false);
+        `,
+      errors: 1,
+    },
+    {
+      code: `
+        var xmlhttp = new ActiveXObject("Microsoft.XMLHTTP");
+        xmlhttp.open("GET", url, false);
```
Comment (vilchik-elena) [inline eslint-bridge/tests/rules/http-requests.test.ts]: add test with different order

File: eslint-bridge/tests/rules/http-requests.test.ts [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+      code: `
+      const request = require('request');
+      request(url, function (error, res, body) {});
+      `,
+      errors: 1,
+    },
+    {
+      code: `
+      const axios = require('axios');
+      axios.post(url);
```
Comment (vilchik-elena) [inline eslint-bridge/tests/rules/http-requests.test.ts]: add test for `axious()`

**Pull request merged**